Change homepage layout

sarinac opened the pull request:
PR for #2098  (update homepage and layout)

The `index.rst` file divides the content into the following sections:
* Section: h1 Overview (note that I had to section it this way in order to use Sphinx's grid feature and get the CSS flexbox working) 
    * section: h2 ArviZ
    * section: h2 Example Gallery
* Section: h1 Key Features - This entire section is a draft. I tossed all the sentences from the existing homepage into this section. (@OriolAbril - feel free to edit this part)
* Section: h1 Support ArviZ
    * div: h3 Contributions
    * div: h3 Citation
    * div: h3 Sponsors
    * div: h3 Donate

LMK what other changes are needed or if the CSS is too messy...(I could try to comment better or reorganize it) 😅 


<!-- readthedocs-preview arviz start -->
----
:books: Documentation preview :books:: https://arviz--2119.org.readthedocs.build/en/2119/

<!-- readthedocs-preview arviz end -->

OriolAbril commented on doc/source/index.rst:
```diff
@@ -2,161 +2,199 @@
+.. grid:: 2 2 3 4
+   :gutter: 3
+
+   .. grid-item-card:: Key Feature 1
```
I like this style of key feature, not sure how it would look with longer titles. I would remove all 

Some of the things we could use:

* Interoperable

      Integrates with all major probabilistic programming libraries: 
      PyMC, CmdStanPy, PyStan, Pyro, NumPyro, emcee...

* Large suite of visualizations

      Provides over 25 plotting functions for all parts of Bayesian workflow:
      visualizing distributions, diagnostics, model checking...
      See the gallery for examples.

* State of the art diagnostics

      Latest published diagnostics and statistics are implemented, tested and distributed
      with ArviZ.

* Flexible model comparison

      Includes functions for comparing models with information criteria,
      and cross validation (both approximate and brute force).

* Built for collaboration

      Designed for flexible cross-language serialization using netCDF or Zarr formats.
      ArviZ also has a [Julia version](link somewhere) that uses the same {ref}`data schema <schema>`

* Labeled data

      Builds on top of xarray to work with labeled dimensions and coordinates

* Open source

      Distributed under the [Apache license](https://github.com/arviz-devs/arviz/blob/main/LICENSE),
      ArviZ is developed and maintained [publicly on GitHub](https://github.com/arviz-devs/arviz) 
      by a vibrant, responsive, and diverse {ref}`community <community>`

---

The last point is taken verbatim from numpy. And those are very clearly a starting point only.

OriolAbril commented on doc/source/index.rst:
```diff
@@ -2,161 +2,199 @@
+
+   <div class="two-col">
+      <!-- Sponsors -->
+      <div class="col" id="sponsors">
```
I might add sponsors, institutional partners and numfocus logo at the bottom with a centered "Sponsors and institutional partners" title and no explanation. 

Then we could do something like:
1. set the content to be wider and use a single 3 element grid for contributing, citing and donating.
2. Use the grid to mention/explain numfocus as we do now and then have 3 "buttons/calls-to-action": donate link like we have now, get ArviZ merchandise ([full logo](https://numfocus.myspreadshop.com/arviz+logo?idea=629e289fc8ee26344a684241) and [favicon logo](https://numfocus.myspreadshop.com/arviz+favicon+design?idea=62a74f17ebe60a221692c6f2) both available) and maybe a 3rd one about becoming a sponsor (we could even write/add a page to arviz.org with sponsor prospectus)

OriolAbril replied:
> Is this layout what you have in mind? 

Option 1 would be restructuring the elements we have now only. We currently have 1 section with 4 squares:

```
## Support ArviZ
contributions                citation
sponsors                     donate
```

they would become

```
## Support ArviZ
contributions      citation      donate

## Sponsors and institutional partners
<-- logos only --!>
```

Example of sponsor section: https://xarray.dev/#sponsors

---

option 2 would be something similar to what we have now:

```
## Support ArviZ
contributions            citation
numfocus (description)   calls to action (donate, buy swag, sponsor)

## Sponsors ...
logos only
```

2x3 grid with more space per element and one button per grid item also sounds good, but all 6 squares would go inside support arviz, then the sponsors section would come.

OriolAbril replied:
> Re: setting content to be wider, the current website and draft has all sections at the same width (see below), but if we have 3 elements per row for this grid then I will likely need to make this specific section wider than the others. I can give it a try to see what works.

The full width text that we currently have inside the key features section has to go, so I would try making everything a bit wider (not full width, maybe not even as wide as the navbar content, but a bit wider). I'd also expect that to be easier and less strange/confusing for readers

OriolAbril reviewed (commented):
let me know what you think about the ideas. I think it is already a huge improvement 🎉

sarinac commented on doc/source/index.rst:
```diff
@@ -2,161 +2,199 @@
+========
 
-ArviZ is a Python package for exploratory analysis of Bayesian models. Includes functions for posterior analysis, data storage, sample diagnostics, model checking, and comparison.
+ArviZ
```
Is "converting the file to markdown" as _straightforward_ as changing the `index.rst` file to `index.md` + changing the file contents to be markdown? If so, I can try that (just wanted to make sure there are not other steps I'm omitting). I think markdown would be friendlier to edit too.

OriolAbril replied:
It is not that straightforward, but if it would help you then I can do it in the coming days. Make sure you have pushed everything and ping me on slack so I can do it without generating git conflicts or loosing work in the process.

sarinac commented on doc/source/index.rst:
```diff
@@ -2,161 +2,199 @@
 
 .. _homepage:
 
-ArviZ: Exploratory analysis of Bayesian models
```
Good catch! I completely missed that. Thanks for linking the documentation too because I wasn't sure how to handle that.

sarinac commented:
Hey @OriolAbril - I attempted to convert to markdown (just used [arviz-project](https://github.com/arviz-devs/arviz-project/blob/main/index.md) as reference), but there are two things I can't figure out:
1. `index.rst` had a `.. _homepage:` reference (line 3), but I don't know how that translates in the new markdown. It's left out right now. 
2. The images in the example gallery are hidden due to `opacity: 0; for the `.example-img-plot-overlay` class in the CSS file. This should have been fine, but the image is nested inside the `example-img-plot-overlay` when it previously wasn't. I don't know how to unnest img in this example below. 
```
      ::::{div} example-img-plot-overlay
      Rank Bars Diagnostic with KDE using `plot_trace`

      :::{image} ./_images/mpl_plot_trace_bars.png
```
Desired output:
```
<div class="example-img-plot-overlay">Rank Bars Diagnostic with KDE using `plot_trace`</div>
<img src="./_images/mpl_plot_trace_bars.png"/>
```

----
Besides the markdown issues...
* Key Features looks okay in terms of word-spacing. I could make the padding bigger, but I'm not clear on what else to do with the style based on the current draft. Some ideas would be including subtle CTAs (e.g. see example gallery would look like "See API Reference ->" button at the top section); including pictures for each box (might be too difficult). 
    * Not exactly Key Features related, but I think a section on how ArviZ is being used by the community would be a good following section. You could link other people's work.
* I think this Sponsors placement looks better than the initial draft.
* The Support Arviz section seems WIP to me still. I think contributing/citing should go before Sponsors because it's not $ related. LMK what you think.
    * Bottom section should conclude with "Donate" (the way it is + merch + future become-a-sponsor). I can add that later this week.

OriolAbril commented:
nesting in rst happened with indexing, in markdown it is closer to html, each directive (grid, div, image...) needs to be both opened and closed. Moreover, nested directives need to have less colons in their fence than their parents.

I have tried to markdownify the file without breaking too many things. I have also removed some indentation as this is markdown and 4 spaces in after an emtpy line is formatted as code so hopefully this will minimize the potential of this. 

> Key Features looks okay in terms of word-spacing. I could make the padding bigger, but I'm not clear on what else to do with the style based on the current draft

I find them a bit hard to read. I would have 3 elements per row (we might be able to delete one item), make it a bit wider or both. I think it is ok to not have any action associated to those, they are mostly things we want visitors to read, not to act on. If it feels too boring we can try to copy numpy: https://numpy.org/ that has a little animation on the grids when you scroll over.

> I think this Sponsors placement looks better than the initial draft.

I have made the logos smaller and placed them at the bottom, but I also agree that having their own section looks better. We don't need to use `grid-item-card` though, we can place them as regular images inside a grid.

sarinac commented:
Thanks for the fixes! I like the use of grid instead of flexbox. 

#### Changes:

* I made a CSS fix on the hero (layout was bad on mobile).
* I re-styled the Key Features section (I hope I don't need to change icons that much 🥺 ), but I think it looks better only have 6 items so I tossed the "Open Source" sentence back to the top of the grid... It doesn't look great, but maybe we can mention it being open source elsewhere? 
    * Re: animation on key features, I don't want to animate anything that's not clickable. Any hover event makes me think I can click on it or get a hover tooltip. So instead, I removed the shadow/border to make it look flat and the opposite of clickable! :D 
* I limited img size of sponsors to 200px so that they won't blow up on a large screen. 

#### To-do:

* What's the reason for putting sponsors at the very bottom? Usually for webpages, the most important things go at the very top (value proposition/hero which we have) and bottom. I see the current priorities of the homepage as 1) Getting Started for Beginners, 2) Discover (AKA example gallery), 3) API documentation for less-beginners, and 4) Donate and these are all CTA-driven. Sponsors is not actionable. I also think the donate button will be easily missed if not placed clearly at the bottom of the page. 
    * I wouldn't worry about Citing or Contributing as priorities/CTAs because the audience will probably come to ArviZ website for the purpose of getting those details. We don't need to emphasize those actions since that audience are likely able to find that info on their own. 
* I think it's better if we split up contributions/citation versus donate/sponsor instead of having them all under "Support ArviZ". However, I don't have good ideas on implementing that yet. I just don't think this current organization is the best...I see it kind of like a catch-all right now 😅 
* Still need to add the additional links in the donate section (tbd, I haven't forgotten!)

OriolAbril commented:
>  I tossed the "Open Source" sentence back to the top of the grid... It doesn't look great, but maybe we can mention it being open source elsewhere? 

We already mention this in other places, so I think it would be best to get rid of this completely instead of adding it as wide text at the begining of the key features section.

> Re: animation on key features, I don't want to animate anything that's not clickable. Any hover event makes me think I can click on it or get a hover tooltip. So instead, I removed the shadow/border to make it look flat and the opposite of clickable! :D

They look amazing, thank you so much!

> What's the reason for putting sponsors at the very bottom? Usually for webpages, the most important things go at the very top (value proposition/hero which we have) and bottom.

Personal experience, I generally don't get to the bottom of the pages I browse so my initial though was to put sponsors there 😅. Very happy to follow better advise and put the donate button at the bottom. 

> I think it's better if we split up contributions/citation versus donate/sponsor instead of having them all under "Support ArviZ". However, I don't have good ideas on implementing that yet. I just don't think this current organization is the best...I see it kind of like a catch-all right now sweat_smile

Maybe the first could be something along the lines of "using arviz?" the other "want to support arviz?" (with better section titles). Or without the centered title and only the 2 subheadings in the columns?

sarinac commented:
> We already mention this in other places, so I think it would be best to get rid of this completely instead of adding it as wide text at the begining of the key features section.

Sounds good. I removed it

> I generally don't get to the bottom of the pages I browse so my initial though was to put sponsors there

Yeah I feel that. If you don't scroll through everything then you're probably not interested or intending to do any of the actions in the Support section anyways. 😅 Usually the edges of the page are more valuable real estate (if you scroll really fast, you'll skip the middle and reach the bottom + people tend to notice and remember first/last items). I put Donate at the bottom for that reason.

> Maybe the first could be something along the lines of "using arviz?" the other "want to support arviz?" (with better section titles). Or without the centered title and only the 2 subheadings in the columns?
I tried "Using ArviZ" > "Sponsors" > "Support ArviZ" and like this structure. I am also down for nesting "Sponsors" under "Support Arviz" (which would just turn it from a `<h1>` to `<h3>` tag), lmk if you prefer that more. 

I also did a little more edits on the descriptions (removed the "TBD/placeholder" texts) but prefer if you check as well! LMK if I'm missing anything else.

OriolAbril commented:
I did one commit to try another variant of the layout, feel free to undo it.

OriolAbril commented on doc/source/_static/custom.css:
```diff
@@ -39,6 +39,143 @@ html[data-theme="dark"] {
+}
+
+/* Sponsors */
+img {
+  max-width: 200px;
+}
```
Just realized this is being applied to _all_ images throughout the website. For example: https://arviz--2119.org.readthedocs.build/en/2119/examples/plot_forest_mixed.html

OriolAbril commented on doc/source/_static/custom.css:
```diff
@@ -39,6 +39,133 @@ html[data-theme="dark"] {
+  margin-bottom: 4rem;
+}
+
+h1.homepage-title .headerlink {
+  visibility: hidden;
+  position: absolute;
+}
```
I would keep this, it is hidden by default but it is nice to easily get a link to specific sections to share.

sarinac replied:
I sorta changed the CSS but tried to keep the header links in the h1 tags (I didn't do it for h3 though). I don't think it was possible to grab the headerlinks before when there was no `id` being generated.

OriolAbril reviewed (commented):
I have messed a couple things, sorry. I will try to fix them. Also added a couple comments.

canyon289 commented:
Very clean. I love this home page!

social media logos should be larger.
![image](https://user-images.githubusercontent.com/7213793/202611180-c3eee439-1acd-4643-90ff-6f5f3e147dd0.png)

We should add mastodon
https://fosstodon.org/@ArviZ@bayes.club

Bokeh looks broken, not sure if its just the preview
![image](https://user-images.githubusercontent.com/7213793/202611264-872c43bd-0595-4562-8370-b220c7160ace.png)


Whitespace between first two "blocks" seems large.
![image](https://user-images.githubusercontent.com/7213793/202611348-b78e7ca1-261e-4c03-a7c3-316f13a00077.png)

The second block is amazing.

OriolAbril commented on doc/source/index.md:
```diff
@@ -0,0 +1,328 @@
+sd_hide_title: true
+---
+
+<!-- CSS overrides on the homepage only -->
```
could we also set width to 70rem here and have it be only for the homepage? I like the homepage a bit wider if there is enough space, especially for the hero element.

OriolAbril commented on doc/source/conf.py:
```diff
@@ -157,17 +157,25 @@
+        #     "url": "https://github.com/arviz-devs/arviz",
+        #     "icon": "fab fa-square-github",
+        # },
+        # {
+        #     "name": "Twitter",
+        #     "url": "https://twitter.com/arviz_devs",
+        #     "icon": "fab fa-square-twitter",
+        # },
         {
-            "name": "GitHub",
-            "url": "https://github.com/arviz-devs/arviz",
-            "icon": "fab fa-github-square",
-        },
-        {
-            "name": "Twitter",
-            "url": "https://twitter.com/arviz_devs",
-            "icon": "fab fa-twitter-square",
-        },
+            "name": "Mastodon",
+            "url": "https://fosstodon.org/@ArviZ@bayes.club",
+            "icon": "fab fa-mastodon",
+        }
     ],
```
How about:

```
"icon_links": [
        {
            "name": "GitHub",
            "url": "https://github.com/arviz-devs/arviz",
            "icon": "fa-brands fa-square",
        },
        {
            "name": "Twitter",
            "url": "https://twitter.com/arviz_devs",
            "icon": "fa-brands fa-twitter",
        },
            "name": "Mastodon",
            "url": "https://bayes.club/@ArviZ",
            "icon": "fa-brands fa-mastodon",
        }
```

which generates:

![imatge](https://user-images.githubusercontent.com/23738400/203581002-1e348b03-0560-46a6-b29f-92ca0e638d8d.png)


If we want to have twitter be blue we can add the css manually. We originally had the shortcuts, but their usage and encouragement on using them from the theme has been irregular, at some point it looked like the shortcuts might even be removed. I feel more comfortable with the explicit icons for now (you might have already noticed the theme is nice but still very new and unstable both look-wise and use-wise)

Side note, I also updated the mastodon link to be directly to the right space.

sarinac replied:
Ooh thanks! This is way better and 100% agree with your comments. I like keeping everything the same color so that one doesn't stand out over the others.

OriolAbril reviewed (commented):
Left two super small comments, I think after that it will be ready to merge, maybe even with no changes

OriolAbril commented:
Thanks, @sarinac! Now we need a release to get the updates out in the stable slug too
